regexp.go

```diff
@@ -422,7 +447,44 @@ restart:
 	return nil
 }
 
+func (r *RegexLexer) compileRules() error {
+	// TODO: refactor to remove this duplication of code from NewLexer
```

Comment:
I think `NewLexer()` might be able to just call `return NewLazyLexer(config, func() Rules{ return rules })`?

Reply:
Great idea.  Done.